lib/p2p/node/supervisor.ex

```diff
@@ -82,23 +82,65 @@ defmodule Elixium.Node.Supervisor do
   # Connects to the bootstrapping peer registry and returns a list of
   # previously connected peers.
   @spec fetch_peers_from_registry(integer) :: List
-  def fetch_peers_from_registry(port) do
+  def fetch_peers_from_registry(port_conf) do
     url = Application.get_env(:elixium_core, :registry_url)
-
-    case :httpc.request(url ++ '/' ++ Integer.to_charlist(port)) do
+    own_local_ip = fetch_local_ip
+    own_public_ip = fetch_public_ip["ip"]
+    case :httpc.request(url ++ '/' ++ Integer.to_charlist(port_conf)) do
       {:ok, {{'HTTP/1.1', 200, 'OK'}, _headers, body}} ->
         peers =
           body
           |> Jason.decode!()
           |> Enum.map(&peerstring_to_tuple/1)
-          |> Enum.filter(fn {_, port} -> port != nil end)
+          |> Enum.filter(fn {peer, port} ->
+            port != nil &&
+            validate_own_ip_port(peer, own_local_ip, port, port_conf) == false &&
+            validate_own_ip_port(peer, own_public_ip, port, port_conf) == false
+          end)
 
         if peers == [], do: :not_found, else: peers
 
       {:error, _} -> :not_found
     end
   end
 
+
+  @doc """
+    On Connection, fetch our public ip
+  """
+  @spec fetch_public_ip :: String.t()
+  def fetch_public_ip do
+   api_url =  'https://api.ipify.org?format=json'
+   case :httpc.request(api_url) do
+     {:ok, {{'HTTP/1.1', 200, 'OK'}, _headers, body}} ->
+         body
+         |> Jason.decode!()
+     {:error, _} -> :not_found
+   end
+  end
+
+  @doc """
+    On Connection, fetch our local ip
+  """
+  @spec fetch_local_ip() :: String.t()
+  def fetch_local_ip do
+    {:ok, [_, {_, ip_list}]} = :inet.getifaddrs()
```

Comment:
This match fails on my mac. Macs have multiple network interfaces, this is what mine looked like:

```elixir
{:ok,
 [
   {'lo0',
    [
      flags: [:up, :loopback, :running, :multicast],
      addr: {127, 0, 0, 1},
      netmask: {255, 0, 0, 0},
      addr: {0, 0, 0, 0, 0, 0, 0, 1},
      netmask: {65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535},
      addr: {65152, 0, 0, 0, 0, 0, 0, 1},
      netmask: {65535, 65535, 65535, 65535, 0, 0, 0, 0}
    ]},
   {'gif0', [flags: [:pointtopoint, :multicast]]},
   {'stf0', [flags: []]},
   {'XHC0', [flags: []]},
   {'XHC20', [flags: []]},
   {'en1',
    [
      flags: [:up, :broadcast, :running, :multicast],
      hwaddr: [194, 0, 64, 226, 107, 1]
    ]},
   {'en2',
    [
      flags: [:up, :broadcast, :running, :multicast],
      hwaddr: [194, 0, 64, 226, 107, 0]
    ]},
   {'en0',
    [
      flags: [:up, :broadcast, :running, :multicast],
      hwaddr: [240, 24, 152, 55, 247, 9],
      addr: {65152, 0, 0, 0, 6383, 57469, 34970, 54065},
      netmask: {65535, 65535, 65535, 65535, 0, 0, 0, 0},
      addr: {192, 168, 1, 184},
      netmask: {255, 255, 255, 0},
      broadaddr: {192, 168, 1, 255}
    ]},
   {'p2p0',
    [
      flags: [:up, :broadcast, :running, :multicast],
      hwaddr: [2, 24, 152, 55, 247, 9]
    ]},
   {'awdl0',
    [
      flags: [:up, :broadcast, :running, :multicast],
      hwaddr: [186, 89, 114, 147, 161, 65],
      addr: {65152, 0, 0, 0, 47193, 29439, 65171, 41281},
      netmask: {65535, 65535, 65535, 65535, 0, 0, 0, 0}
    ]},
   {'bridge0',
    [
      flags: [:up, :broadcast, :running, :multicast],
      hwaddr: [194, 0, 64, 226, 107, 1]
    ]},
   {'utun1',
    [
      flags: [:up, :pointtopoint, :running, :multicast],
      addr: {65152, 0, 0, 0, 16263, 12641, 17902, 48462},
      netmask: {65535, 65535, 65535, 65535, 0, 0, 0, 0},
      dstaddr: {20, 18, 15, 0}
    ]},
   {'utun0',
    [
      flags: [:up, :pointtopoint, :running, :multicast],
      addr: {65152, 0, 0, 0, 53184, 28, 11637, 51778},
      netmask: {65535, 65535, 65535, 65535, 0, 0, 0, 0},
      dstaddr: {53, 54, 57, 57}
    ]}
 ]}
```